src/Mollie/WC/Helper/PaymentMethodsIconUrl.php

```diff
@@ -39,7 +62,130 @@ protected function fallToAssets($paymentMethodName)
             return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . 's.svg');
         }
 
-        return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . '.svg');
+        return Mollie_WC_Plugin::getPluginUrl('public/images/' . $paymentMethodName . self::SVG_FILE_EXTENSION
+        );
+    }
+
+    /**
+     * Method that handles the composition of the new composed svg file
+     * and returns its url
+     *
+     * @param array $enabledCreditCards Array with the enabled creditcards
+     *
+     * @return string the url to the composed svg
+     */
+    protected function composeSvgImage($enabledCreditCards)
+    {
+        $this->writeToSvgFile();
+
+        $actual = $this->buildSvgComposed($enabledCreditCards);
+
+        return  $this->writeToSvgFile($actual);
+    }
+
+    /**
+     * Is the customization of credit card icons enabled?
+     *
+     * @return bool
+     */
+    protected function enabledCreditcardOptions()
+    {
+        $creditcardSettings = get_option(
+            'mollie_wc_gateway_creditcard_settings'
+        );
+        if (!isset($creditcardSettings['mollie_creditcard_icons_enabler'])) {
+            return false;
+        }
+        return wc_string_to_bool(
+            $creditcardSettings['mollie_creditcard_icons_enabler']
+        );
+    }
```

Comment:
Do we really need this method? 

If you encapsulate the logic of `enabledCreditcardOptions()` into `enabledCreditcards()` you can let the latter return an empty array, then in `composeSvgImage` you just return an empty string for failure or you throw (which actually is missed because `enabledCreditcards()` can still return an empty array if there are no cards enabled, see line 117).

The thing is that if you call `enabledCreditcards()` directly you can still get a list of enabled credit cards even if `enabledCreditcardOptions()` return false, but you'll not know that because you may call just `enabledCreditcards()`, this will end up in an error. If you couple the two logics you'll not need to always have a condition to check if the settings are enabled, you just ask to retrieve the enabled credit cards and you get a list if there are any. Nothing else, just remember to call one method, the one you want the things from.

Therefore, my impression is that encapsulating the `enabledCreditcardOptions()` within `enabledCreditcards()` it's the better approach, and return empty array if it's the case.

Check it out

```
protected function enabledCreditcards()
{
    // TODO 'mollie_creditcard_icons_enabler' should be a constant
    $creditCardSettings = get_option('mollie_wc_gateway_creditcard_settings', false) ?: [];
    $enabled = isset($creditCardSettings['mollie_creditcard_icons_enabler'])
        ? wc_string_to_bool($creditCardSettings['mollie_creditcard_icons_enabler'])
        : false;
        
    if (!$enabled) {
        return [];
    }    
    
    $optionLexem = self::MOLLIE_CREDITCARD_ICONS;
    $creditcardsAvailable = self::AVAILABLE_CREDITCARD_ICONS;
    $svgFileName = self::SVG_FILE_EXTENSION;
    $enabledCreditcards = [];

    foreach ($creditcardsAvailable as $card) {
        if (!isset($creditCardSettings[$optionLexem . $card])) {
            continue;
        }
        
        if (wc_string_to_bool($creditCardSettings[$optionLexem . $card])) {
            $enabledCreditcards[] = $card . $svgFileName;
        }
    }

    return $enabledCreditcards;
}
```